modules/core/src/main/scala/org/tessellation/domain/cluster/programs/TrustPush.scala

```diff
@@ -1,30 +1,57 @@
 package org.tessellation.domain.cluster.programs
 
 import cats.effect.Async
+import cats.syntax.applicative._
 import cats.syntax.flatMap._
 import cats.syntax.functor._
 
-import org.tessellation.domain.trust.storage.TrustStorage
+import org.tessellation.domain.trust.storage.{SnapshotOrdinalTrustStorage, TrustStorage}
+import org.tessellation.schema.SnapshotOrdinal
 import org.tessellation.sdk.domain.gossip.Gossip
 
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.NonNegative
+import eu.timepit.refined.refineV
+import eu.timepit.refined.types.numeric.NonNegLong
+
 object TrustPush {
 
   def make[F[_]: Async](
     trustStorage: TrustStorage[F],
+    ordinalTrustStorage: SnapshotOrdinalTrustStorage[F],
```

Comment:
btw, why `trustStorage` cannot store trust by ordinal? IMO there is no need to split that functionality between two storages. both are trust

Reply:
`TrustStorage` holds the real-time trust information. It is the trust ratings of a peer as known by the node at that point in time. It is updated in real time. These values are gossiped as they are received. `SnapshotOrdinalTrustStorage` holds the trust values for peers as of an ordinal. It is not updated in real-time. These values are gossiped only once in a while.

Reply:
> These values are gossiped as they are received.
so it's not real time

anyway, you can store both in the same storage. also, if you already gossip the trust, why can't you add an ordinal and store it once per n ordinals?

Looks like you have:
First service: store every trust (gossiped as just trust)
Second service: store trust every N ordinal (gossiped as trust and ordinal)

instead, you can do:
One service (gossip trust always with ordinal, as it determines a point in time):
 - store every trust (ignore ordinal)
 - store trust every N ordinal (take ordinal into account)

Reply:
It was implemented this way because in a discussion ion with @sdavidp and @ryle-goehausen , we decided that two storages would be easier to implement right now.

Reply:
What's harder in writing the same methods in a different file?